### ISSUE TITLE
Unable to run application after migrate to nx v17.2.5

### ISSUE THREAD

**Issue by selvadhoni** [nrwl/nx#20830]
### Current Behavior

>  NX   Successfully finished running migrations from 'migrations.json'. This workspace is up to date!

Done in 148.37s.

after migrate when i try to run my application im getting below errors.

$ nx run my-app:serve --verbose

> nx run my-app:serve

 >  NX   Cannot read properties of undefined (reading 'build')

TypeError: Cannot read properties of undefined (reading 'build')
    at readTargetOptions (D:\btm-version-upgrade\4.0\atb-gmt\node_modules\@nx\devkit\src\executors\read-target-options.js:21:61)
    at nodeExecutor (D:\btm-version-upgrade\4.0\atb-gmt\node_modules\@nx\js\src\executors\node\node.impl.js:37:43)
    at nodeExecutor.next (<anonymous>)
    at getLastValueFromAsyncIterableIterator (D:\btm-version-upgrade\4.0\atb-gmt\node_modules\nx\src\utils\async-iterator.js:13:27)        
    at iteratorToProcessStatusCode (D:\btm-version-upgrade\4.0\atb-gmt\node_modules\nx\src\command-line\run\run.js:41:94)
    at D:\btm-version-upgrade\4.0\atb-gmt\node_modules\nx\src\command-line\run\run.js:155:16
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async handleErrors (D:\btm-version-upgrade\4.0\atb-gmt\node_modules\nx\src\utils\params.js:9:16)
    at async process.<anonymous> (D:\btm-version-upgrade\4.0\atb-gmt\node_modules\nx\bin\run-executor.js:59:28)

### Expected Behavior

after successfull migrate nx latest, i need to run my application without any error. So what should i do to run my applicatin successfully.

### Steps to Reproduce

step 1: yarn install

step 2: nx migrate latest

step 3: yarn nx migrate --run-migrations

step 4: npx nx run my-app:serve

after this command error below will occur

$ nx run my-app:serve --verbose

> nx run my-app:serve

 >  NX   Cannot read properties of undefined (reading 'build')

TypeError: Cannot read properties of undefined (reading 'build')
    at readTargetOptions (D:\btm-version-upgrade\4.0\atb-gmt\node_modules\@nx\devkit\src\executors\read-target-options.js:21:61)
    at nodeExecutor (D:\btm-version-upgrade\4.0\atb-gmt\node_modules\@nx\js\src\executors\node\node.impl.js:37:43)
    at nodeExecutor.next (<anonymous>)
    at getLastValueFromAsyncIterableIterator (D:\btm-version-upgrade\4.0\atb-gmt\node_modules\nx\src\utils\async-iterator.js:13:27)        
    at iteratorToProcessStatusCode (D:\btm-version-upgrade\4.0\atb-gmt\node_modules\nx\src\command-line\run\run.js:41:94)
    at D:\btm-version-upgrade\4.0\atb-gmt\node_modules\nx\src\command-line\run\run.js:155:16
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async handleErrors (D:\btm-version-upgrade\4.0\atb-gmt\node_modules\nx\src\utils\params.js:9:16)
    at async process.<anonymous> (D:\btm-version-upgrade\4.0\atb-gmt\node_modules\nx\bin\run-executor.js:59:28)

### Nx Report

```shell
$ nx report

 >  NX   Report complete - copy this into the issue template

   Node   : 20.10.0
   OS     : win32-x64
   yarn   : 1.22.19

   nx (global)        : 17.2.6
   nx                 : 17.2.6
   @nx/js             : 17.2.6
   @nx/jest           : 17.2.6
   @nx/linter         : 17.2.6
   @nx/eslint         : 17.2.6
   @nrwl/linter       : 12.7.1
   @nx/workspace      : 17.2.6
   @nx/devkit         : 17.2.6
   @nx/eslint-plugin  : 17.2.6
   @nx/node           : 17.2.6
   @nrwl/tao          : 17.2.6
   typescript         : 4.3.5
   ---------------------------------------
   The following packages should match the installed version of nx
     - @nrwl/linter@12.7.1

   To fix this, run `nx migrate nx@17.2.6`
```

### Package Manager Version

npm -v 10.2.5

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by jaysoo** [comment]
Which version are you migrating from? Can you also remove `@nrwl/tao` and `@nrwl/linter`? `@nx/linter` should replace `@nrwl/linter`, and the version should make the other Nx packages (`17.2.6`).

Please try running `nx reset` as well before trying serve again.

If you have a reproduction, that would help a lot in investigating this issue. Otherwise, can you paste you `project.json` file so we can inspect it?

**Comment by selvadhoni** [comment]
Hi @jaysoo, Thanks for the replay. My old version is `@nrwl - v12.7.1`. I removed `@nrwl/tao` and `@nrwl/linter` as well.

this is my one of the project.json file, since i have many project.json files for other many services.

{
  "name": "api-setup",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "apps/api-setup/src",
  "projectType": "application",
  "targets": {
    "build": {
      "executor": "@nx/js:node:build",
      "outputs": ["{options.outputPath}"],
      "options": {
        "generatePackageJson": true,
        "outputPath": "dist/apps/api-setup",
        "main": "apps/api-setup/src/main.ts",
        "tsConfig": "apps/api-setup/tsconfig.app.json",
        "assets": [
          {
            "input": "libs/core/report/src/asset",
            "glob": "**/*",
            "output": "./assets"
          }
        ]
      },
      "configurations": {
        "production": {
          "optimization": true,
          "extractLicenses": true,
          "inspect": false,
          "fileReplacements": [
            {
              "replace": "apps/api-setup/src/environments/environment.ts",
              "with": "apps/api-setup/src/environments/environment.prod.ts"
            }
          ]
        }
      }
    },
    "serve": {
      "executor": "@nx/js:node:execute",
      "options": {
        "buildTarget": "api-setup:build"
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint"
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["coverage/apps/api-setup"],
      "options": {
        "jestConfig": "apps/api-setup/jest.config.js"
      }
    }
  }
}

 **in this file i changed two lines manually:**

1. build section `"executor": "@nx/node:build"` this to `"executor": "@nx/js:node:build"`
2. serve section `"executor": "@nx/node:execute"` this to `"executor": "@nx/js:node:execute"`

**when i try without these changes i getting this error:**

$ nx serve api-setup --verbose

 >  NX   Unable to resolve @nx/node:execute.

   The "@nx/node" package does not support Nx executors.

Error: Unable to resolve @nx/node:execute.
The "@nx/node" package does not support Nx executors.
    at getExecutorInformation (D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\src\command-line\run\executor-utils.js:53:15)
    at getExecutorForTask (D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\src\tasks-runner\utils.js:188:56)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async getCustomHasher (D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\src\tasks-runner\utils.js:192:22)
    at async D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\src\hasher\hash-task.js:12:30
    at async Promise.all (index 0)
    at async hashTasksThatDoNotDependOnOutputsOfOtherTasks (D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\src\hasher\hash-task.js:11:30)
    at async invokeTasksRunner (D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\src\tasks-runner\run-command.js:126:5)
    at async D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\src\tasks-runner\run-command.js:89:24
    at async handleErrors (D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\src\utils\params.js:9:16)

**after i changed then getting this error:**

$ nx serve api-setup --verbose

> nx run api-setup:serve


 >  NX   Cannot read properties of undefined (reading 'build')


TypeError: Cannot read properties of undefined (reading 'build')
    at readTargetOptions (D:\btm-version-upgrade\5.0\atb-gmt\node_modules\@nx\devkit\src\executors\read-target-options.js:21:61)
    at nodeExecutor (D:\btm-version-upgrade\5.0\atb-gmt\node_modules\@nx\js\src\executors\node\node.impl.js:37:43)
    at nodeExecutor.next (<anonymous>)
    at getLastValueFromAsyncIterableIterator (D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\src\utils\async-iterator.js:13:27)
    at iteratorToProcessStatusCode (D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\src\command-line\run\run.js:41:94)
    at D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\src\command-line\run\run.js:155:16
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async handleErrors (D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\src\utils\params.js:9:16)
    at async process.<anonymous> (D:\btm-version-upgrade\5.0\atb-gmt\node_modules\nx\bin\run-executor.js:59:28)

 ——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————— 

 >  NX   Running target serve for project api-setup failed

   Failed tasks:

   - api-setup:serve

   Hint: run the command with --verbose for more details.

finally fin the nx report below.

$ nx report

 >  NX   Report complete - copy this into the issue template

   Node   : 20.10.0
   OS     : win32-x64
   yarn   : 1.22.19

   nx (global)        : 17.2.6
   nx                 : 17.2.6
   @nx/js             : 17.2.6
   @nx/jest           : 17.2.6
   @nx/linter         : 17.2.6
   @nx/eslint         : 17.2.6
   @nrwl/linter       : 12.7.1
   @nx/workspace      : 17.2.6
   @nx/devkit         : 17.2.6
   @nx/eslint-plugin  : 17.2.6
   @nx/node           : 17.2.6
   @nrwl/tao          : 17.2.6
   typescript         : 4.3.5
   ---------------------------------------
   The following packages should match the installed version of nx
     - @nrwl/linter@12.7.1

   To fix this, run `nx migrate nx@17.2.6`
